src/mraa.c

```diff
@@ -277,21 +277,31 @@ mraa_deinit()
             }
             free(sub_plat);
         }
+
+        int i = 0;
 #if defined(JSONPLAT)
         if (plat->platform_type == MRAA_JSON_PLATFORM) {
             // Free the platform name
             free(plat->platform_name);
             plat->platform_name = NULL;
+        }
 
-            int i = 0;
-            // Free the UART device path
+        // Free the UART device path
+        for (i = 0; i < plat->uart_dev_count; i++) {
+            if (plat->uart_dev[i].device_path != NULL) {
+                free(plat->uart_dev[i].device_path);
+            }
+        }
+#endif
+        // Free the UART device path for UP2
+        if (strncmp((plat->platform_name), "UP2", strlen("UP2")+1) == 0) {
```

Comment:
1. This duplication of code for freeing paths is inefficient. Now that you have at least two platform types which need this to be done, let's just remove the piece under `#if defined(JSONPLAT)` and simply do the freeing ` if ((plat->platform_type == MRAA_JSON_PLATFORM ) || (plat->platform_type == MRAA_UP2))`;
1. Using platform types enum makes the condition shorter, so let's use it instead of string-comparing the platform name;

Reply:
done, using the enum now.